Sources/ArcGISToolkit/Components/Popups/PopupView.swift

```diff
@@ -203,6 +206,24 @@ extension PopupView {
 }
 
 extension PopupView {
+    /// The visibility of the popup title.
+    /// - Since: 200.6
+    public enum TitleVisibility: Sendable {
+        /// The popup title is hidden.
+        case hidden
+        /// The popup title is visible.
+        case visible
+    }
+    
+    /// Specifies the visibility of the popup title.
+    /// - Parameter visibility: The visibility of the popup title.
+    /// - Since: 200.6
+    public func popupTitle(_ visibility: TitleVisibility) -> Self {
```

Comment:
I'm not set on this modifier for one reason - the `showCloseButton` button modifier.

It seems disjoint that a developer can hide the title and simultaneously leave the close button visible.
```swift
PopupView(popup: popup!, isPresented: $showPopup)
    .popupTitle(.hidden)
    .showCloseButton(true)
```
<img width="374" alt="Screenshot 2024-08-14 at 10 56 03 AM" src="https://github.com/user-attachments/assets/1c370ab2-5ce1-4b02-a028-9cc77b4575f2">


Should we instead deprecate `showCloseButton` and add something like `.popupHeader(_:)` which would hide both?

Reply:
The title and close button are not necessarily related, other than they have been occupying the same space. It may be that the client doesn't want the title in the popup/view but still requires the close button. So I'm OK having them both be distinct and put the onus on the client to figure out how they want it to display/behave.

Reply:
I agree with David that it is a bit disjoint. I'm not sure we should've had a simple boolean for the close button. Some people might prefer a "Done" button. There is no way to customize that (well I guess the library is open source so people _could_ customize it). 

The `isPresented` works if using the toolkit's FloatingPanel. And I guess it works with `.sheet` as well, but it's actually unnecessary with `.sheet` because we could have used the DismissAction environment value:

```swift
@Environment(\.dismiss) private var dismiss
```

I think, for the most part, our components should not assume their container. 

Since title text is sometimes (often) on the "same line" as some buttons (done, cancel, menu), we might do better with a constructor like this that enables the PopupView to build a header for situations where they want to add a close button, but the title isn't displayed in a navigation toolbar:

```swift
struct PopupView<HeaderContent: View>: View {
    let popup: Popup
    let headerContent: ((String) -> HeaderContent)?

    ...
    
    /// Creates a PopupView...
    /// - Parameters:
    ///   - popup: The Popup to display
    ///   - headerContent: A closure that builds the header for a given title
    init(popup: Popup, @ViewBuilder headerContent: @escaping (String) -> HeaderContent) {
        self.popup = popup
        self.headerContent = headerContent
    }
```

This would allow somebody to add their own title, buttons, etc.

That would solve the close button thing. Then I guess there could be some modifier for `header(_ visibility:)`.

Reply:
@mhdostal Any preference on the individual title and close button controls vs. Ryan's suggestion?

Reply:
I'm OK with @rolson 's suggestion. That does solve the issue with the "close" button and allows clients to set it up how they want. It's a good point that our components should not assume their container. We could be better to trying different containers to verify that.